mlflow/llama_index/__init__.py

```diff
@@ -0,0 +1,385 @@
+
+    return Model.log(
+        artifact_path=artifact_path,
+        engine_type=engine_type,
+        engine_kwargs=engine_kwargs,
+        flavor=mlflow.llama_index,
+        registered_model_name=registered_model_name,
+        index=index,
+        conda_env=conda_env,
+        code_paths=code_paths,
+        signature=signature,
+        input_example=input_example,
+        await_registration_for=await_registration_for,
+        pip_requirements=pip_requirements,
+        extra_pip_requirements=extra_pip_requirements,
+        metadata=metadata,
+        **kwargs,
+    )
+
+
+def _save_model(index, path):
+    """Serialize settings, additional llama index objects, and the index."""
+    index_path = os.path.join(path, _INDEX_PERSIST_FOLDER)
+    index.storage_context.persist(persist_dir=index_path)
+
+
+def _load_model(path, flavor_conf):
+    """Deserialize settings, additional llama index objects, and the index."""
+    _add_code_from_conf_to_system_path(path, flavor_conf)
+
+    index_path = os.path.join(path, _INDEX_PERSIST_FOLDER)
+    storage_context = StorageContext.from_defaults(persist_dir=index_path)
+    return load_index_from_storage(storage_context)
+
+
+def load_model(model_uri, dst_path=None):
+    """
+    Load a ``llama_index`` index from a local file or a run.
+
+    Args:
+        model_uri: The location, in URI format, of the MLflow model. For example:
+
+            - ``/Users/me/path/to/local/model``
+            - ``relative/path/to/local/model``
+            - ``s3://my_bucket/path/to/model``
+            - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+            - ``mlflow-artifacts:/path/to/model``
+
+            For more information about supported URI schemes, see
+            `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
+            artifact-locations>`_.
+        dst_path: The local filesystem path to utilize for downloading the model artifact.
+            This directory must already exist if provided. If unspecified, a local output
+            path will be created.
+
+    Returns:
+        A ``llama_index`` index object.
+    """
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
+    flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
+
+    settings_path = os.path.join(local_model_path, _SETTINGS_DIRECTORY)
+    # Settings is a singleton, so it does not need to be passed
+    _ = deserialize_json_to_settings(settings_path)
+
+    return _load_model(local_model_path, flavor_conf)
+
+
+def _load_pyfunc(path):
+    engine_kwargs_path = os.path.join(path, _ENGINE_KWARGS_DIRECTORY)
+    engine_kwargs = deserialize_json_to_engine_kwargs(engine_kwargs_path)
+
+    flavor_conf = _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME)
+    engine_type = flavor_conf["engine_type"]
+    return _LlamaIndexModelWrapper(_load_model(path, flavor_conf), engine_type, engine_kwargs)
```

Comment:
extract engine_type and engine_kwargs from model_config by popping. What other args are available for instantiating that we might use model_config kwargs entries for? How will these be handled?